python/whylogs/core/schema.py

```diff
@@ -76,15 +86,18 @@ def __post_init__(self) -> None:
                 LARGE_CACHE_SIZE_LIMIT,
             )
 
-        if self.types:
-            for col, tpe in self.types.items():
-                self._columns[col] = ColumnSchema(
-                    dtype=tpe, resolver=self.resolvers, type_mapper=self.type_mapper, cfg=self.default_configs
-                )
+        for col, tpe in self.types.items():
+            self._columns[col] = ColumnSchema(
+                dtype=tpe, resolver=self.resolvers, type_mapper=self.type_mapper, cfg=self.default_configs
+            )
 
     def copy(self) -> "DatasetSchema":
         """Returns a new instance of the same underlying schema"""
-        copy = DatasetSchema(**self.__dict__)
+        key_dict = vars(self).copy()
```

Comment:
Personally I think this is more clear :)

Reply:
The new changes to be more clear.

Reply:
We'll have to update all the examples, test code, and documentation to stop subclassing `DatasetSchema`.

Reply:
ooh how fun